libindy/src/services/wallet/mod.rs

```diff
@@ -63,22 +64,30 @@ impl<'a> JsonDecodable<'a> for WalletConfig {}
 #[derive(Debug)]
 pub struct WalletCredentials {
     master_key: [u8; 32],
+    rekey: Option<Box<[u8; 32]>>,
```

Comment:
Is boxing really required here?

Reply:
I dont want to increase the size of WalletCredentials by 32 bytes since 99% of the time it will be empty

Reply:
@jovfer 
@keichiri 

I see the following problems here:

- 32 is a magic number
- Memory allocated for Keys should be zeroed after usage. See any sodiumoxide Key type
- Box here looks like non-obvious micro-optimization.

Reply:
This (except the Box) is the same approach used for master key in credentials. 
So I think if it is an issue now, it must have been before, since it was implemented.

About Box, why is it nonobvious? It is totally normal practice to use memory smartly?

Ok, i will change all of this as soon as possible

Reply:
agree with Janko about the 2nd point from Slava's review. It's something like out of scope of this PR. This issue was introduced in first implementation of encrypted wallet and wasn't fixed while latest large refactoring from Wallet to Wallet+Storage.
As for Box, this is really micro-optimization. WalletCredential used as single object. 24 bytes vs minor improvement of readability - I prefer doesn't think why Box is needed here, but don't used for the first filed of the structure

Reply:
I removed Box. IT was not used for first field since it is mandatory, always present.

But I agree its not necessary so I removed it.

Anyway, about 2nd point. I agree that it is very needed, and I will open an additional PR soon for it.

I will push the change now